### ISSUE TITLE
Serializing RequestObject

### ISSUE THREAD

**Issue by cryptoquick** [kardeiz/jsonrpc-v2#17]
So, I'm trying to fudge my way into having a jsonrpc_v2-based RPC client, since jsonrpsee is pretty terrible.

Here's what I have so far:

```rust
pub async fn call_rpc_method<T>(method_name: &str) -> Result<T, surf::Error> {
    let url = env::var(API_INFO_KEY).unwrap_or(DEFUALT_URL.to_owned());
    let rpc_call = jsonrpc_v2::RequestObject::request()
        .with_method(method_name)
        .finish();
    let http_res = surf::post(url)
        .body(surf::Body::from_bytes(rpc_call.to_bytes()))
        // .body(surf::Body::from_json(&rpc_call)?)
        .await?;

    http_res.body_json().await
}
```

Just sketching out some ideas.

I ran into an error, however, since this is not a use-case jsonrpc-v2 is designed for. Is there a way to serialize a RequestObject from RequestBuilder?

If not, I would be glad to provide a PR with a client. I'd also provide an implementation with authorization header support, and tokio and async-std implementations. I'm also investigating a type-safe way to describe the API interface. What are your suggestions, @kardeiz ?

**Comment by kardeiz** [comment]
Hi @cryptoquick,

Do you mean, is there a way to serialize `jsonrpc_v2::RequestObject` into a string/bytes? If so, I think we could just add `#[derive(Serialize)]` to `RequestObject`, since all of the field impl `Serialize`.

Do you mean, you would provide a `jsonrpc_v2::Client` object? Depending on what it looked like, I think that would be nice. I haven't really followed `jsonrpsee` or the other JSON-RPC libraries, so I don't really know what the current best practices are for describing API interfaces.

**Comment by cryptoquick** [comment]
Yep, both sound good! If you add the Serialize trait, I'll contribute a Client once I come up with a good approach.

**Comment by kardeiz** [comment]
Okay, I've published the `Serialize` change in `0.10.1`.

**Comment by cryptoquick** [comment]
Much appreciated! Closing.
